plugin-api/beta/client/hocs/withTags.js

```diff
@@ -0,0 +1,178 @@
+        }
+      }
+    `, {
+      props: ({mutate}) => ({
+        addTag: ({id, name, assetId}) => {
+          return mutate({
+            variables: {
+              id,
+              name,
+              asset_id: assetId
+            },
+            optimisticResponse: {
+              addTag: {
+                __typename: 'ModifyTagResponse',
+                errors: null,
+              }
+            },
+            update: (proxy) => {
+              const fragmentId = `Comment_${id}`;
+
+              // Read the data from our cache for this query.
+              const data = proxy.readFragment({fragment: COMMENT_FRAGMENT, id: fragmentId});
+
+              data.tags.push({
+                tag: {
+                  __typename: 'Tag',
+                  name: Tag.toUpperCase()
+                },
+                __typename: 'TagLink'
+              });
+
+              // Write our data back to the cache.
+              proxy.writeFragment({fragment: COMMENT_FRAGMENT, id: fragmentId, data});
+            },
+          });
+        }}),
+    });
+
+  const withRemoveTag = withMutation(
+    gql`
+      mutation RemoveTag($id: ID!, $asset_id: ID!, $name: String!) {
+        removeTag(tag: {name: $name, id: $id, item_type: COMMENTS, asset_id: $asset_id}) {
+          ...ModifyTagResponse
+        }
+      }
+    `, {
+      props: ({mutate}) => ({
+        removeTag: ({id, name, assetId}) => {
+          return mutate({
+            variables: {
+              id,
+              name,
+              asset_id: assetId
+            },
+            optimisticResponse: {
+              removeTag: {
+                __typename: 'ModifyTagResponse',
+                errors: null,
+              }
+            },
+            update: (proxy) => {
+              const fragmentId = `Comment_${id}`;
+
+              // Read the data from our cache for this query.
+              const data = proxy.readFragment({fragment: COMMENT_FRAGMENT, id: fragmentId});
+
+              const idx = data.tags.findIndex((i) => i.tag.name === Tag.toUpperCase());
+
+              data.tags = [...data.tags.slice(0, idx), ...data.tags.slice(idx + 1)];
+
+              // Write our data back to the cache.
+              proxy.writeFragment({fragment: COMMENT_FRAGMENT, id: fragmentId, data});
+            }
+          });
+        }}),
+    });
+  
+  class WithTags extends React.Component {
+
+    postTag = () => {
+      const {comment, asset} = this.props;
+      
+      this.props.addTag({
+        id: comment.id,
+        name: Tag.toUpperCase(),
+        assetId: asset.id
+      });
+    }
+
+    deleteTag = () => {
+      const {comment, asset} = this.props;
+
+      this.props.removeTag({
+        id: comment.id,
+        name: Tag.toUpperCase(),
+        assetId: asset.id
+      });
+    }
+    
+    render() {
+      const {comment} = this.props;
+
+      const isTagged = (tags) => !!tags.filter((t) => t.tag.name === Tag.toUpperCase()).length;
```

Comment:
Functions like this get recreated on render, potentially hurting performance a lot or a little depending on how high the component is in the tree, and whether it is passed into props. This is a hoc, so be careful.

See my general comment for more reading.